src/frametree.cpp

```diff
@@ -577,6 +577,7 @@ void FrameTree::replaceNode(shared_ptr<HSFrame> old,
     auto parent = old->getParent();
     if (!parent) {
         assert(old == root_);
+        // replacement->parent needs to be nulled?
```

Comment:
Yes, definitely! The assertion `!(root_->parent)` must always hold (and is violated here).

Reply:
Added the assertion

Reply:
Maybe I should have said it more explicitly: to fix the issue, I think we need to make the assertion hold, i.e. add
```
root_->parent = {};
```
instead of the assertion

Reply:
D'oh. Ok,changed it, now it works. Thank you Thorsten!

Should I remove the test? Or is there a way to get the geometry of a frame, so I can verify it in the test?

Reply:
Updated the test